services/src/main/java/org/apache/druid/cli/CliCoordinator.java

```diff
@@ -173,6 +174,7 @@ protected List<? extends Module> getModules()
     List<Module> modules = new ArrayList<>();
 
     modules.add(JettyHttpClientModule.global());
+    modules.add(new JoinableFactoryModule());
```

Comment:
why was this needed?

Reply:
It is used to determine whether the datasource is joinable in `PhysicalDatasourceMetadataBuilder`